plugins/gtav/gtav.cpp

```diff
@@ -3,118 +3,213 @@
+static procptr_t memory_offset = 0;
+static std::string identity;
+
+mumble_error_t mumble_init(uint32_t) {
+	return MUMBLE_STATUS_OK;
+}
+
+void mumble_shutdown() {
+}
+
+MumbleStringWrapper mumble_getName() {
+	static const char name[] = "Grand Theft Auto V";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = name;
+	wrapper.size           = strlen(name);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+mumble_version_t mumble_getAPIVersion() {
+	return MUMBLE_PLUGIN_API_VERSION;
+}
+
+void mumble_registerAPIFunctions(void *) {
+}
+
+void mumble_releaseResource(const void *) {
+}
+
+mumble_version_t mumble_getVersion() {
+	// we reuse the GTA version as plugin version
+	return { 1, 54, 2245 };
+}
+
+MumbleStringWrapper mumble_getAuthor() {
+	static const char author[] = "MumbleDevelopers";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = author;
+	wrapper.size           = strlen(author);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+MumbleStringWrapper mumble_getDescription() {
+	static const char description[] = "Provides positional audio functionality for GTA V. "
+									  "Supports GTA V version 1.54 Build 2245 (Steam) and 1.38 (Retail)."
+									  "Also provides identity based on username.";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = description;
+	wrapper.size           = strlen(description);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+uint32_t mumble_getFeatures() {
+	return MUMBLE_FEATURE_POSITIONAL;
+}
+
+uint8_t mumble_initPositionalData(const char *const *programNames, const uint64_t *programPIDs, size_t programCount) {
+	auto ret                = MUMBLE_PDEC_ERROR_TEMP;
+	constexpr auto exe_name = "GTA5.exe";
+
+	for (size_t i = 0; i < programCount; ++i) {
+		if (strcmp(programNames[i], exe_name) != 0) {
+			continue;
+		}
+
+		proc = std::make_unique< ProcessWindows >(static_cast< procid_t >(programPIDs[i]), programNames[i]);
+		const auto &modules = proc->modules();
+		const auto iter     = modules.find(exe_name);
+		if (iter == modules.cend()) {
+			break;
+		}
+		// find out the offset of the memory region
+		const auto &regions = iter->second.regions();
+		const auto &r       = regions.cbegin();
+		if (r != regions.cend()) {
+			memory_offset = r->address;
+		}
+
+		// Check if we can get meaningful data from it
+		float apos[3], afront[3], atop[3], cpos[3], cfront[3], ctop[3];
+		const char *sidentity = "";
+		const char *scontext  = "";
+
+		if (mumble_fetchPositionalData(apos, afront, atop, cpos, cfront, ctop, &scontext, &sidentity)) {
+			ret = MUMBLE_PDEC_OK;
+		}
+
+		break;
+	}
+
+	return ret;
+}
+
+void mumble_shutdownPositionalData() {
+	proc.reset();
+}
+
+bool mumble_fetchPositionalData(float *avatarPos, float *avatarDir, float *avatarAxis, float *cameraPos,
+								float *cameraDir, float *cameraAxis, const char **contextPtr,
+								const char **identityPtr) {
 	for (int i = 0; i < 3; i++) {
```

Comment:
Since the plugin is basically rewritten from scratch anyway, I would change this to:

```suggestion
	for (uint8_t i = 0; i < 3; ++i) {
```